src/Application/GameConfig.h

```diff
@@ -81,67 +76,71 @@ namespace Application {
-              * 0 - Vanilla behaviour, items will be lost.
-              * 1 - Try to place previously non-fit items on every chest opening.
-              * 2 - Try to place previously non-fit items on every item pickup from the chest. */
-            ConfigValue<int> ChestTryPlaceItems = ConfigValue<int>(this, "chest_try_place_items", 2);
+            Int ArtifactLimit = Int(this, "artifact_limit", 13, &ValidateArtifactLimit,
+                                    "Max number of artifacts that the game can generate as loot in any one playthrough. Use 0 for unlimited.");
 
-            /** Maximum allowed slack for point-inside-a-polygon checks when calculating floor z level.
-              * This is needed because there are actual holes in level geometry sometimes, up to several units wide. */
-            ConfigValue<int> FloorChecksEps = ConfigValue<int>(this, "floor_checks_eps", 3, &ValidateFloorChecksEps);
+            Int ChestTryPlaceItems = Int(this, "chest_try_place_items", 2,
+                                         "Fix problems with item loss in high-level chests. "
+                                         "Use 0 for vanilla behaviour, items that don't fit will be lost. "
+                                         "Use 1 to try to place items that didn't fit every time the chest is opened again. "
+                                         "Use 2 to try to place items that didn't fit every an item is picked up from the chest.");
 
-            /** Gravity strength, the higher the more gravity, 0 - disable gravity completely. */
-            ConfigValue<int> Gravity = ConfigValue<int>(this, "gravity", 5);
+            Int FloorChecksEps = Int(this, "floor_checks_eps", 3, &ValidateFloorChecksEps,
+                                     "Maximum allowed slack for point-inside-a-polygon checks when calculating floor z level. "
+                                     "This is needed because there are actual holes in level geometry sometimes, up to several units wide.");
 
-            /** Maximum depth for item pickup / opening chests / activating levers / etc with a keyboard (by pressing trigger key). */
-            ConfigValue<float> KeyboardInteractionDepth = ConfigValue<float>(this, "keyboard_interaction_depth", 512.0f, &ValidateInteractionDepth);
+            Int Gravity = Int(this, "gravity", 5, "Gravity strength, the higher the more gravity, 0 disables gravity completely.");
 
-            /** Minimum recovery time for melee weapons. 30 - vanilla */
-            ConfigValue<int> MinRecoveryMelee = ConfigValue<int>(this, "minimum_recovery_melee", 30, &ValidateRecovery);
+            Float KeyboardInteractionDepth = Float(this, "keyboard_interaction_depth", 512.0f, &ValidateInteractionDepth,
+                                                   "Maximum range for item pickup / opening chests / activating levers / etc "
+                                                   "with a keyboard (by pressing spacebar).");
```

Comment:
we should use a action or description name like trigger/interaction key as any specific key could be rebinded